src/v/kafka/server/tests/produce_partition_bench.cc

```diff
@@ -0,0 +1,164 @@
+
+#include <tuple>
+
+static ss::logger fpt_logger("fpt_test");
+
+using namespace std::chrono_literals; // NOLINT
+
+struct produce_partition_fixture : redpanda_thread_fixture {
+    static constexpr size_t topic_name_length = 30;
+    static constexpr size_t total_partition_count = 1;
+
+    model::topic t;
+
+    produce_partition_fixture() {
+        BOOST_TEST_CHECKPOINT("before leadership");
+
+        wait_for_controller_leadership().get0();
+
+        BOOST_TEST_CHECKPOINT("HERE");
+
+        t = model::topic(
+          random_generators::gen_alphanum_string(topic_name_length));
+        auto tp = model::topic_partition(t, model::partition_id(0));
+        add_topic(
+          model::topic_namespace_view(model::kafka_namespace, t),
+          total_partition_count)
+          .get();
+
+        auto ntp = make_default_ntp(tp.topic, tp.partition);
+        wait_for_leader(ntp).get0();
+        BOOST_TEST_CHECKPOINT("HERE");
+    }
+
+    future<> run_single_partition(void) {
+        BOOST_TEST_CHECKPOINT("HERE");
+
+        model::topic_partition tp = model::topic_partition(
+          t, model::partition_id(0));
+
+        storage::record_batch_builder builder(
+          model::record_batch_type::raft_data, model::offset{0});
+
+        constexpr size_t num_batches = 1;
+
+        for (size_t i = 0; i < num_batches; ++i) {
+            builder.add_raw_kv(iobuf{}, iobuf{});
+        }
+
+        auto batch = std::move(builder).build();
+
+        std::vector<kafka::produce_request::partition> partitions;
+        partitions.push_back(kafka::produce_request::partition{
+          .partition_index{model::partition_id(0)},
+          .records = kafka::produce_request_record_data(std::move(batch))});
+
+        std::vector<kafka::produce_request::topic> topics;
+        topics.push_back(kafka::produce_request::topic{
+          .name{std::move(tp.topic)}, .partitions{std::move(partitions)}});
+
+        std::optional<ss::sstring> t_id;
+        auto acks = -1;
+        kafka::produce_request produce_req = kafka::produce_request(
+          t_id, acks, std::move(topics));
+
+        auto conn = make_connection_context();
+
+        BOOST_TEST_CHECKPOINT("HERE");
+
+        kafka::request_header header{
+          .key = kafka::produce_handler::api::key,
+          .version = kafka::produce_handler::max_supported};
+
+        // Use a fake_req to generate the context because encoding the request
+        // steals the request's iobufs.
+        kafka::produce_request fake_req;
+        auto rctx = make_request_context(fake_req, header, conn);
+
+        BOOST_TEST_CHECKPOINT("HERE");
+
+        kafka::produce_ctx pctx{
+          std::move(rctx),
+          std::move(produce_req),
+          kafka::produce_response{},
+          ss::default_smp_service_group()};
+
+        std::vector<ss::future<>> dispatched;
+        std::vector<ss::future<kafka::produce_response::partition>> produced;
+
+        perf_tests::start_measuring_time();
```

Comment:
I am confused, this is now being run 10k times in different fibers for a single benchmark loop?

Reply:
Yeah good point, this isn't going to work correctly.

Since my original motiviation for cranking up the concurrency was to get a useful flamegraph, we can just remove the concurrency and simplify things.